tests/test_rdscluster.py

```diff
@@ -736,6 +736,25 @@ def test_clear_permissions(self):
         )
         self.assertEqual(len(restore_permissions_after), 0)
 
+    def test_pending_maintenance(self):
+        session_factory = self.replay_flight_data("test_rdscluster_pending_maintenance")
+        p = self.load_policy(
+            {
+                "name": "rds-cluster-pending-maintenance",
+                "resource": "rds-cluster",
+                "filters": [
+                    {
+                        "type": "pending-maintenance"
+                    }
+                ],
+            },
+            config={"region": "us-west-2"},
+            session_factory=session_factory,
+        )
+
+        resources = p.run()
+        self.assertEqual(len(resources), 0)
```

Comment:
@sgungi-intuit shouldn't this match something?

Reply:
for example, why are we not seeing a match for `docdb-2022-01-13-21-04-08`?

Reply:
@sgungi-intuit I see, as there is pending maintenance for the DBs but not the cluster.... maybe best to have a test case that is a able to match when a cluster does have pending maintenance?

Reply:
@sgungi-intuit I see- only the DB instances have pending maintenance, but not the RDS clusters themselves.

Probably best to have a test case with at least one cluster that does have pending maintenance to ensure the filter works as expected when there are matching resources

Reply:
fixed the test case and added logic to pick arn instead of cluster id.